### PR TITLE
Lighting_Engine: Align to Luminaire oM Orientation Change

### LINKED ISSUE #3037: Lighting_Engine: Update Lighting methods for change to Orientation on Luminaire
#### Description:
As per the title.

### PR DESCRIPTION
**Pull request by enarhi**
### NOTE: Depends on 
https://github.com/BHoM/BHoM/pull/1501

   
### Issues addressed by this PR

Closes #3037 

Changed Luminaire and Transform methods to accommodate new Orientation property.

### Test files
[test files](https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/03_Alpha/BHoM/BHoM_Engine/Lighting_Engine?csf=1&web=1&e=QXqYyt)

### CODE REVIEW

**Comment by enarhi** [comment]
@BHoMBot check compliance

**Comment by enarhi** [comment]
@BHoMBot check core

**Comment by enarhi** [comment]
@BHoMBot check unit-tests
@BHoMBot check null-handling

**Comment by enarhi** [comment]
@BHoMBot check unit-tests

**Comment by enarhi** [comment]
@BHoMBot check compliance

**Review by JohnSloane — DISMISSED** [review]
All appears to be working as expected. I tested this with the test files in rhino and gh to ensure all orientation is working appropriately with all engine methods.

File: Lighting_Engine/Create/Luminaire.cs [hunk 1]
```diff
@@ -35,6 +35,8 @@
 
 using BH.Engine.Geometry;
 using BH.Engine.Spatial;
+using BH.Engine.Base;
+using ICSharpCode.Decompiler.IL.Transforms;
```
Comment (FraserGreenroyd) [inline Lighting_Engine/Create/Luminaire.cs:39]: Why has this using been added?

File: Lighting_Engine/Create/Luminaire.cs [hunk 2]
```diff
@@ -156,10 +158,29 @@ public static List<Luminaire> Luminaire(this ICurve crv, int count, Vector dir,
         [Output("luminaire", "The created luminaire.")]
         public static Luminaire Luminaire(this Point pt, Vector orientation = null, LuminaireType type = null, string name = "")
         {
+            Basis basis = null;
+
+            if (orientation != null && orientation.Length() != 0)
+            {
+                double angle = orientation.Angle(Vector.ZAxis);
+                {
```
Comment (FraserGreenroyd) [inline Lighting_Engine/Create/Luminaire.cs:166]: Why is this bit in its own brackets?

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

File: Lighting_Engine/Create/Luminaire.cs (hunk 2 above)
Comment (enarhi) [reply Lighting_Engine/Create/Luminaire.cs:169]: fixed

File: Lighting_Engine/Create/Luminaire.cs (hunk 1 above)
Comment (enarhi) [reply Lighting_Engine/Create/Luminaire.cs]: removed

**Comment by enarhi** [comment]
@BHoMBot check versioning

**Comment by enarhi** [comment]
@BHoMBot check code-compliance
@BHoMBot check copyright-compliance
@BHoMBot check project-compliance

**Comment by enarhi** [comment]
@BHoMBot check documentation-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check null-handling
@BHoMBot check serialisation
@BHoMBot check unit-tests

**Review by FraserGreenroyd — APPROVED** [review]
Last commit only fixed my changes, functionality review from @JohnSloane still valid.

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: dataset-compliance

**Pull request merged**